addons/account/tests/test_account_move_tax_lock_date.py

```diff
@@ -0,0 +1,110 @@
+from odoo.addons.account.tests.account_test_classes import AccountingTestCase
+from odoo.exceptions import ValidationError, UserError
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+from calendar import monthrange
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class TestTaxBlockDate(AccountingTestCase):
+    """
+    Forbid creation, edition and deletion of Journal Items with a date prior
+    to the Tax Block Date
+    """
+
+    def setUp(self):
+        super(TestTaxBlockDate, self).setUp()
+        self.user_id = self.env.user
+        
+        last_day_month = datetime.now()
+        last_day_month = last_day_month.replace(day=monthrange(last_day_month.year, last_day_month.month)[1])
+        self.last_day_month_str = last_day_month.strftime(DEFAULT_SERVER_DATE_FORMAT)
+        first_day_month = datetime.now()
+        first_day_month = first_day_month.replace(day=1)
+        self.first_day_month_str = first_day_month.strftime(DEFAULT_SERVER_DATE_FORMAT)
+        middle_day_month = datetime.now()
+        middle_day_month = middle_day_month.replace(day=15)
+        self.middle_day_month_str = middle_day_month.strftime(DEFAULT_SERVER_DATE_FORMAT)
+
+        self.sale_journal_id = self.env['account.journal'].search([('type', '=', 'sale')])[0]
+        self.account_id = self.env['account.account'].search([('internal_type', '=', 'receivable')])[0]
+        self.other_account_id = self.env['account.account'].search([('internal_type', '!=', 'receivable')])[0]
+        
+        self.move = {
+            'name': '/',
+            'journal_id': self.sale_journal_id.id,
+            'date': self.middle_day_month_str,
+            'line_ids': [(0, 0, {
+                    'name': 'foo',
+                    'debit': 10,
+                    'account_id': self.account_id.id,
+                }), (0, 0, {
+                    'name': 'bar',
+                    'credit': 10,
+                    'account_id': self.account_id.id,
+                })]
+        }
```

Comment:
I don't get how your tests are supposed to work as you don't give any tax_ids or tax_line_id

Reply:
The secret is that the tests didn't work